docs/guide/extending-matchers.md

```diff
@@ -26,13 +26,11 @@ expect.extend({
 If you are using TypeScript, you can extend default `Assertion` interface in an ambient declaration file (e.g: `vitest.d.ts`) with the code below:
 
 ```ts
-import type { Assertion, AsymmetricMatchersContaining } from 'vitest'
-
 interface CustomMatchers<R = unknown> {
   toBeFoo: () => R
 }
 
-declare module 'vitest' {
+declare module '@vitest/expect' {
```

Comment:
This is not the correct way to augment it. `@vitest/expect` is an implementation detail and should not be exposed. Testing Library gives a bad example

Reply:
aggre about `export {}` though in my case it wasn't needed since my `global.d.ts` file anyway imports some of mine types.

regarding `vitest` instead of `@vitest/expect` I played around a little bit more and here what I found:

-  both types from here `import type { Assertion, AsymmetricMatchersContaining } from 'vitest'` are considered unused and hence cleaned up by linters/formatters
- without importing something from 'vitest` `declare module 'vitest'` doesn't work for me

so what do you think about the following example? no `export` is needed since there is `import` and `@vitest/expect` is hidden as internal implementation detail

```ts
import 'vitest';

interface CustomMatchers<R = unknown> {
  toBeFoo: () => R
}

declare module 'vitest' {
  interface Assertion<T = any> extends CustomMatchers<T> {}
  interface AsymmetricMatchersContaining extends CustomMatchers {}
}
```

Reply:
Sure, looks good to me 👍🏻